### PR TITLE
report: Add macbox7 GSoC report

### PR DESCRIPTION
**Pull request by MacBox7**
https://deploy-preview-614--coala-projects.netlify.com/gsoc/2018/statusreport/macbox7

### CODE REVIEW

**Comment by jayvdb** [comment]
https://deploy-preview-614--coala-projects.netlify.com/GSoC/2018/StatusReport/macbox7 is 404

File: reports/macbox7.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+  - Gitter:
+    - username: macbox7
+    - link: https://gitter.im/macbox7
+
+email: ajankit2304@gmail.com
+
+blog: https://macbox7.github.io/Blog-GSOC18/
+
+activity:
+- 0:
```
Comment (jayvdb) [inline reports/macbox7.md]: these need to be indented.

we dont have a bear for this ;-(

File: reports/macbox7.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+   that can be used to traverse pyflakes AST of the input python module.
+5. Establish the proof of concept by creating `NoFutureImportBear` that
+   detects and fixes any usage of `__future__` imports in a python file.
+6. Add handlers for `ARGS` and `ARGUMENTS` and provide the functionality
+   of partial code analysis in `pyflakes`.
+7. Create a `PEP8NamingBear` that detects all pep8 naming violations
+   in a python file. Special handling is also done to detect violations
+   in doctests which other similar plugins fail to perform.
+8. Create a `BuiltinsBear` that is capable of detecting redefinition
+   of python builtins in a file.
```
Comment (jayvdb) [inline reports/macbox7.md]: file -> module

File: reports/macbox7.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+traversing logic and need to have much no more than the knowledge of
+PyFlakesASTBear nodes to write the bears. However, designing the logic
+first by writing cEP(coala enhancement proposal) for
+[coala-pyflakes](https://github.com/coala/cEPs/blob/master/cEP-0025.md) helped
+me a lot to understand the problem to a deep level and come out with a logic
+to tackle it.
+
+#### Future prospects of the projects
+
+With `PyFlakesASTBear` ready bear developers can use it to detect codesmells
```
Comment (jayvdb) [inline reports/macbox7.md]: "ready" isnt clear.  Is it ready now?  If so, using "with .. ready" is redundant, at best.

Also migrating flake8 plugins to this new infrastructure should be easy tasks that people can now work on.

And more importantly, we had planned for this bear framework to be able to load & run flake8 ast plugins as-is, so porting wasnt necessary; that can be mentioned as a medium to large sized project which is now feasible.

File: reports/macbox7.md [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+blog: https://macbox7.github.io/Blog-GSOC18/
+activity:
+  - 0:
+    - repo: coala-pyflakes
+    - link: https://gitlab.com/coala/bears/coala-pyflakes/commit/684f1a958dd66fa781e7f03777ffa96b7e6a2ca7
+    - details: nitial commit
+  - 1:
+    - repo: coala-pyflakes
+    - link: https://gitlab.com/coala/bears/coala-pyflakes/commit/9484d8af901153c490812d86809760b14c46d9af
+    - details: PyFlakesASTBear.py: Add PyFlakesASTBear
```
Comment (virresh) [inline reports/macbox7.md]: Use quotes or `:` will be treated as defining yaml variable

File: reports/macbox7.md [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+  - Gitter:
+    - username: macbox7
+    - link: https://gitter.im/macbox7
+email: ajankit2304@gmail.com
+blog: https://macbox7.github.io/Blog-GSOC18/
+activity:
+  - 0:
+    - repo: coala-pyflakes
+    - link: https://gitlab.com/coala/bears/coala-pyflakes/commit/684f1a958dd66fa781e7f03777ffa96b7e6a2ca7
+    - details: nitial commit
```
Comment (virresh) [inline reports/macbox7.md]: Probably a type

**Comment by jayvdb** [comment]
More commits at https://github.com/coala/coala/commits?author=MacBox7 and https://github.com/coala/coala-bears/commits?author=MacBox7

**Comment by jayvdb** [comment]
Also https://gitlab.com/coala/PyPrint/graphs/master and https://gitlab.com/coala/mobans/graphs/master and https://gitlab.com/coala/coala-utils/graphs/master , and maybe others on gitlab ;)

File: reports/macbox7.md [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+   - details: >
+       Handle subtrees as input to checker
+ - 51:
+   - repo: pyflakes
+   - link: https://github.com/PyCQA/pyflakes/commit/fb1f223ac2141772f140be6f1108d02009c1d658
+   - details: >
+       checker.py: Handle UnboundLocal error for builtins
+ - 52:
+   - repo: pyflakes
+   - link: https://github.com/PyCQA/pyflakes/pull/329
```
[the remark this answers is not in the document]
Comment (jayvdb) [reply reports/macbox7.md]: merged

File: reports/macbox7.md [hunk 7]
```diff
@@ -0,0 +1,489 @@
+---
+layout: report
+permalink: /GSoC/2018/StatusReport/macbox7
```
Comment (jayvdb) [inline reports/macbox7.md]: remove this and layout:

add date, categories, etc, as shown at https://github.com/coala/projects/pull/633/files#diff-f1164fdbd5814b6d270a59c0f9e3939e

File: reports/macbox7.md [hunk 8]
```diff
@@ -0,0 +1,492 @@
...
+organisation: coala
+organisation_link: https://coala.io
+project: Integrate pyflakes-enhanced AST into coala
+project_link: https://summerofcode.withgoogle.com/projects/#5549789140221952
+tarball: https://rawgit.com/MacBox7/GSoC-Report/master/project.tar.gz
+mentors: >
+ [John Vandenberg](https://github.com/jayvdb) & [Adhika Setya Pramudita](https://github.com/adhikasp) & [Steven Myint](https://github.com/myint)
+
+phase:
+ - Bonding : https://gitlab.com/coala/GSoC/GSoC-2018/milestones/8
```
Comment (jayvdb) [inline reports/macbox7.md]: fix intendation as recently done in the example.

use yamllint to verify

File: reports/macbox7.md [hunk 9]
```diff
@@ -0,0 +1,489 @@
...
+organisation_link: https://coala.io
+project: Integrate pyflakes-enhanced AST into coala
+project_link: https://summerofcode.withgoogle.com/projects/#5549789140221952
+tarball: https://rawgit.com/MacBox7/GSoC-Report/master/project.tar.gz
+mentors: >
+ [John Vandenberg](https://github.com/jayvdb),
+ [Adhika Setya Pramudita](https://github.com/adhikasp),
+ [Steven Myint](https://github.com/myint)
+phase:
+ - Bonding: https://gitlab.com/coala/GSoC/GSoC-2018/milestones/8
```
Comment (jayvdb) [inline reports/macbox7.md]: yamllint.

**Comment by jayvdb** [comment]
Also needs a rebase

**Comment by jayvdb** [comment]
Still not rebased, and still not fixed https://github.com/coala/projects/pull/614#discussion_r209581406

File: _reports/macbox7.md [hunk 10]
```diff
@@ -0,0 +1,489 @@
...
+organisation_link: https://coala.io
+project: Integrate pyflakes-enhanced AST into coala
+project_link: https://summerofcode.withgoogle.com/projects/#5549789140221952
+tarball: https://rawgit.com/MacBox7/GSoC-Report/master/project.tar.gz
+mentors: >
+ [John Vandenberg](https://github.com/jayvdb),
+ [Adhika Setya Pramudita](https://github.com/adhikasp),
+ [Steven Myint](https://github.com/myint)
+phase:
+ - Bonding: https://gitlab.com/coala/GSoC/GSoC-2018/milestones/8
```
Comment (jayvdb) [inline _reports/macbox7.md]: yamllint

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Pull request merged**